### DISCUSSION TITLE
docs/3.x/api-reference/checkout/mutations/checkout-shipping-method-update

### DISCUSSION THREAD

**Discussion by giscus** [saleor/saleor-docs#964] [General]
# docs/3.x/api-reference/checkout/mutations/checkout-shipping-method-update

Update shipping address in the existing checkout.

https://docs.saleor.io/docs/3.x/api-reference/checkout/mutations/checkout-shipping-method-update

**Comment by selbieh** [comment]
i got this error message 
"errors": [
                {
                    "field": "shippingAddress",
                    "message": "This checkout doesn't need shipping",
                    "code": "SHIPPING_NOT_REQUIRED"
                }
            ]
when i try to mutate the shippingAddress

**Reply by afonsofigs** [reply]
Hi!
The products on the checkout you are trying to mutate are probably set as non-shippable items or don't have any shipping methods associated to them.
Before mutating the shipping address you must make sure the checkout has the "isShippingRequired" boolean flag set to true. If this is set to false, then just skip this mutation and mutate the billingAddress.
